python/cuml/manifold/umap.pyx

```diff
@@ -194,6 +197,9 @@ class UMAP(Base):
                 feature is made optional in the GPU version due to the
                 significant overhead in copying memory to the host for
                 computing the hash. (default = False)
+    random_state : int, RandomState instance or None, optional (default=None)
+        random_state is the seed used by the random number generator during
+        embedding initialization and during sampling used by the optimizer
```

Comment:
Given my response about the separate code path, I think we should make the note in here that enabling this will produce completely deterministic results, but at the expense of making synchronous updates to the gradients during optimization, which will slightly increase memory usage.